src/main.ts

```diff
@@ -14,7 +14,14 @@ async function run(): Promise<void> {
   info(`Previous version: ${previousVersion}`);
   setOutput('previous-version', previousVersion);
 
-  await checkout(getEnv('GITHUB_REF'));
+  let checkoutRef;
+  try {
+    checkoutRef = getEnv('GITHUB_HEAD_REF') || getEnv('GITHUB_REF');
+  } catch {
+    checkoutRef = getEnv('GITHUB_REF');
+  }
```

Comment:
So, just a friendly comment, but may be able to do this a bit cleaner as follows:

```ts
// GITHUB_HEAD_REF is needed to support PRs
const checkoutRef = getEnvOrUndefined('GITHUB_HEAD_REF') || getEnv('GITHUB_REF');
```

And then just make `getEnvOrUndefined` as a copy of `getEnv` in `utils.ts` that doesn't throw the error at the end.